packages/paste-website/functions/opengraph.ts

```diff
@@ -0,0 +1,114 @@
+// Shout outs to the following repositories:
+
+// https://github.com/whitep4nth3r/puppeteer-demo
+// https://github.com/vercel/og-image
+// https://github.com/ireade/netlify-puppeteer-screenshot-demo
+
+// The maximum execution timeout is 10
+// seconds when deployed on a Personal Account (Hobby plan).
+// For Teams, the execution timeout is 60 seconds (Pro plan)
+// or 900 seconds (Enterprise plan).
+
+import puppeteer from 'puppeteer-core';
+import chromium from 'chrome-aws-lambda';
+import type {Handler} from '@netlify/functions';
+
+/** The code below determines the executable location for Chrome to
+ * start up and take the screenshot when running a local development environment.
+ *
+ * If the code is running on Windows, find chrome.exe in the default location.
+ * If the code is running on Linux, find the Chrome installation in the default location.
+ * If the code is running on MacOS, find the Chrome installation in the default location.
+ * You may need to update this code when running it locally depending on the location of
+ * your Chrome installation on your operating system.
+ */
+const PlatformPath = {
+  win32: 'C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe',
+  linux: '/usr/bin/google-chrome',
+  darwin: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome',
+};
+const exePath = PlatformPath[process.platform] || PlatformPath.darwin;
+
+async function getOptions(isDev: boolean): Promise<{args: any; executablePath: any; headless: any}> {
+  if (isDev) {
+    return {
+      args: [],
+      executablePath: exePath,
+      headless: true,
+    };
+  }
+
+  return {
+    executablePath: await chromium.executablePath,
+    args: chromium.args,
+    headless: chromium.headless,
+  };
+}
+
+const handler: Handler = async (event) => {
+  const componentRequested = event.path.replace('/.netlify/functions/opengraph/', '');
+  // console.log('component requested:', componentRequested);
```

Comment:
Yeah, I tend to keep some logging for node things, because they are running on a server you don't get a web inspector to play with, only a console in Netlify. Just kind of useful to see what it's doing when it's invoked. Ideally it'd pass things to a logging service like datadog but that seems a little extra

Reply:
Yeah I kind of left them commented because this type of thing is tedious to debug if someone happens, so they serve as kind of comments to help debug anything in the future.